common/test/acceptance/tests/lms/test_bookmarks.py

```diff
@@ -276,24 +295,146 @@ def test_unreachable_bookmark(self):
+        And previous and next buttons are disabled
+        """
+        self._test_setup(num_chapters=2)
+        self._bookmark_units(num_units=2)
+
+        self.bookmarks_page.click_bookmarks_button()
+        self.assertTrue(self.bookmarks_page.results_present())
+        self._verify_pagination_info(
+            bookmark_count_on_current_page=2,
+            header_text='Showing 1-2 out of 2 total',
+            previous_button_enabled=False,
+            next_button_enabled=False,
+            current_page_number=1,
+            total_pages=1
+        )
+
+    def test_pagination_with_multiple_pages(self):
+        """
+        Scenario: Bookmarks list pagination is working as expected for multiple pages
+        Given that I am a registered user
+        And I visit my courseware page
+        And I have bookmarked all the 12 units available
+        Then I click on Bookmarks button
+        And I should see a bookmarked list of 10 items
+        And I should see paging header and footer with correct data
+        And previous page button is disabled in footer
+        And next page button is enabled in footer
+        Then I click on next page button in footer
+        And I should be navigated to second page
+        And I should see a bookmarked list with 2 items
+        And I should see paging header and footer with correct data
+        And next page button is disabled in footer
+        And previous page button is enabled in footer
+        Then I click on previous page button
+        And I should be navigated to first page
+        And I should see paging header and footer with correct data
+        """
+        self._test_setup(num_chapters=12)
+        self._bookmark_units(num_units=12)
+
+        self.bookmarks_page.click_bookmarks_button()
+        self.assertTrue(self.bookmarks_page.results_present())
+
+        self._verify_pagination_info(
+            bookmark_count_on_current_page=10,
+            header_text='Showing 1-10 out of 12 total',
+            previous_button_enabled=False,
+            next_button_enabled=True,
+            current_page_number=1,
+            total_pages=2
+        )
+
+        self.bookmarks_page.click_paging_footer_button('next')
+
+        self._verify_pagination_info(
+            bookmark_count_on_current_page=2,
+            header_text='Showing 11-12 out of 12 total',
+            previous_button_enabled=True,
+            next_button_enabled=False,
+            current_page_number=2,
+            total_pages=2
+        )
+
+        self.bookmarks_page.click_paging_footer_button('previous')
+
+        self._verify_pagination_info(
+            bookmark_count_on_current_page=10,
+            header_text='Showing 1-10 out of 12 total',
+            previous_button_enabled=False,
+            next_button_enabled=True,
+            current_page_number=1,
+            total_pages=2
+        )
+
+    def test_pagination_with_page_number(self):
```

Comment:
This tests two distinct behaviors: successfully changing to a valid page, and being prevented from changing to an invalid page. Please split it up into those two respective test cases.